File: typescript/src/payment/integration/paymentRequestIntegrationCommand.ts
// Published Language esposta dal BC payment per richiedere un nuovo pagamento.
// Usa solo tipi stabili (string, Money) in modo da essere invocabile
// da qualsiasi bounded context senza introdurre coupling sui tipi interni di payment.

import type { Money } from '@/common/domain/primitive/money.js';
import { requireArgument } from '@/common/utils/requireArgument.js';

export type PaymentRequestIntegrationCommand = {
  readonly kind: 'PaymentRequestIntegrationCommand';
  readonly clientReference: string;
  readonly amount: Money;
};

export function paymentRequestIntegrationCommand(
  clientReference: string,
  amount: Money
): PaymentRequestIntegrationCommand {
  requireArgument(clientReference, 'clientReference');
  requireArgument(amount, 'amount');
  return { kind: 'PaymentRequestIntegrationCommand', clientReference, amount };
}

File: java/src/main/java/it/giannibombelli/wsc2026/payment/integration/PaymentRequestIntegrationCommand.java
package it.giannibombelli.wsc2026.payment.integration;

import it.giannibombelli.wsc2026.common.domain.primitive.Money;
import it.giannibombelli.wsc2026.common.utils.Require;

/**
 * Published Language di {@code payment} per richiedere un nuovo pagamento,
 * senza esporre il modello interno del BC ai consumatori ({@code booking}, {@code giftcard}).
 */
public record PaymentRequestIntegrationCommand(String clientReference, Money amount) {
    public PaymentRequestIntegrationCommand {
        Require.requireArgument(clientReference, "clientReference");
        Require.requireArgument(amount, "amount");
    }
}
